AllisonsCatPalaceTDDTests/LocationControllerTests.swift

```diff
@@ -0,0 +1,332 @@
+    }
+
+    func testZipCodeFieldCharacterLimit() {
+        textField.text = ""
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "8"))
+
+        textField.text = "8"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 1, length: 0), replacementString: "0"))
+
+        textField.text = "80"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 2, length: 0), replacementString: "2"))
+
+        textField.text = "802"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 3, length: 0), replacementString: "2"))
+
+        textField.text = "8022"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 4, length: 0), replacementString: "0"))
+
+        textField.text = "80220"
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 5, length: 0), replacementString: "8"))
+    }
+
+    func testZipCodeFieldBeginsEditing() {
+        // resets borderColor
+        textField.layer.borderColor = UIColor.green.cgColor
+        XCTAssertEqual(textField.layer.borderColor, UIColor.green.cgColor,
+                       "Zip code field's border color should be green")
+
+        delegate.textFieldDidBeginEditing?(textField)
+        XCTAssertEqual(textField.layer.borderColor, UIColor.lightGray.cgColor,
+                       "Zip code field's border color should be reset to light gray on beginning text editing")
+
+        // stops spinner
+        controller.activityIndicator.startAnimating()
+        XCTAssertTrue(controller.activityIndicator.isAnimating,
+                      "Activity indicator should be animating")
+        delegate.textFieldDidBeginEditing?(textField)
+        XCTAssertFalse(controller.activityIndicator.isAnimating,
+                       "Activity indicator should stop animating on beginning text editing")
+    }
+
+    func testZipCodeFieldEndsEditingAtFiveCharacters() {
+        UITextField.ResignFirstResponderSpyController.createSpy(on: textField)?.spy {
+            // Need to be in the window to becomeFirstResponder
+            replaceRootViewController(with: controller)
+            textField.becomeFirstResponder()
+            XCTAssertTrue(textField.isEditing,
+                          "Textfield should be editing when it's the first responder")
+
+            textField.text = "80220"
+            controller.textFieldDidChange(textField)
+
+            XCTAssertFalse(textField.isEditing,
+                           "Textfield should not be editing when it is not the first responder")
+            XCTAssertTrue(textField.resignFirstResponderCalled,
+                          "Textfield should resign first responder at five characters")
+        }
+    }
+
+    func testZipCodeFieldResignsFirstResponderWhenDoneEditing() {
```

Comment:
no reason for this since didEndEditing is called when first responder is resigned.